exercises/allergies/canonical-data.json

```diff
@@ -19,107 +69,338 @@
           "expected": false
         },
         {
-          "description": "cats",
+          "description": "allergic only to peanuts",
           "property": "allergicTo",
           "input": {
-            "item": "cats",
+            "item": "peanuts",
+            "score": 2
+          },
+          "expected": true
+        },
+        {
+          "description": "allergic to peanuts and something else",
+          "property": "allergicTo",
+          "input": {
+            "item": "peanuts",
+            "score": 7
+          },
+          "expected": false
```

Comment:
This is confusing.  Why wouldn't it be true?  Doing this in my head, 7 means, eggs, peanuts and shellfish right?  Are we supposed to expect score to be exactly 2 for this to be true?

Reply:
How about, `"allergic to peanuts and nothing else"`?

Reply:
I guess the next case does that.  I must just be biased because I am too familiar with the 1.2.0 version of the exercise.  The changes make the test data seem more convoluted to me.  That doesn't mean there is something wrong with it.  If I had not been predisposed I would probably look at this and shrug and say it is probably OK.

Sorry I am not very helpful.  I do not use a test generator so can only review these changes in terms of how they help me to process the information.  With that said what was here before was fine, but I can see how automatic test generation might be challenged and would benefit from breaking down the tests into a simpler form.  I think that is the goal is it not?

Reply:
The intent was they are the same tests in the original, but rather than putting them all together in one clump - you present them like someone doing TDD and introduce them one after another. So yes the goal is that a generator can infer the tests to generate, rather than having 1 large test that pasts a huge dictionary and gives a weird looking API to the class you are testing.

All the other language tracks have taken that large expectation and manually cut it up into single tests - so I'm proposing we do what they did up front in the spec.

In doing this, it has exposed us all to asking questions about the text could be worded differently - but its actually the same assertions as was there originally (is my claim, anyway)

Reply:
@rpottsoh 
> This is confusing. Why wouldn't it be true? Doing this in my head, 7 means, eggs, peanuts and shellfish right? Are we supposed to expect score to be exactly 2 for this to be true?

There's a good reason why it's confusing: it *is* incorrect. I had written a quick script for generating the new cases, and I had entered the wrong expectation for generating this test. I've corrected the issue.

Reply:
Guys - I'm sorry this opened a can of worms... but it is better if they can be automated (but also read in situ).